src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java

```diff
@@ -330,6 +349,70 @@ private void parseJabRefComment(Map<String, String> meta) {
 
             // custom entry types are always re-written by JabRef and not stored in the file
             dumpTextReadSoFarToString();
+        } else if (comment.startsWith(MetaData.BIBDESK_STATIC_FLAG)) {
+            parseBibDeskComment(comment, meta);
+        }
+    }
+
+    /**
+     * Adds BibDesk group entries to the JabRef database
+     * */
+    private void addBibDeskGroupEntriesToJabRefGroups() {
+        for (String groupName: parsedBibdeskGroups.keySet()) {
+            String[] citationKeys = parsedBibdeskGroups.get(groupName).split(",");
+            for (String citation : citationKeys) {
+                if (database.getEntryByCitationKey(citation).isPresent()) {
+                    String groupValue = database.getEntryByCitationKey(citation).get().getField(StandardField.GROUPS).orElse(null);
+                    if (groupValue == null) { // if the citation does not belong to a group already
+                        database.getEntryByCitationKey(citation).get().setField(StandardField.GROUPS, groupName);
+                    } else { // if the citation does belong to a group already, we concatinate
+                        groupValue += "," + groupName;
+                        database.getEntryByCitationKey(citation).get().setField(StandardField.GROUPS, groupValue);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Parses comment types found in BibDesk, to migrate BibDesk Static Groups to JabRef.
+     * */
+    private void parseBibDeskComment(String comment, Map<String, String> meta) {
+        String xml = comment.substring(MetaData.BIBDESK_STATIC_FLAG.length() + 1, comment.length() - 1);
+        try {
+            // Build a document to handle the xml tags
+            DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
```

Comment:
Extract this factor creation as a property variable because creating the factory is an expensive task